theatre/core/src/projects/DefaultAssetStorage.ts

```diff
@@ -0,0 +1,172 @@
+import * as idb from 'idb-keyval'
+import type Project from './Project'
+import {val} from '@theatre/dataverse'
+import type {IAssetStorageConfig} from './Project'
+// @ts-ignore
+import blobCompare from 'blob-compare'
+
+function getAllPossibleAssetIDs(project: Project) {
+  const sheets = Object.values(val(project.pointers.historic.sheetsById))
+  const staticValues = sheets
+    .flatMap((sheet) => Object.values(sheet?.staticOverrides.byObject ?? {}))
+    .flatMap((overrides) => Object.values(overrides ?? {}))
+  const keyframeValues = sheets
+    .flatMap((sheet) => Object.values(sheet?.sequence?.tracksByObject ?? {}))
+    .flatMap((tracks) => Object.values(tracks?.trackData ?? {}))
+    .flatMap((track) => track?.keyframes)
+    .map((keyframe) => keyframe?.value)
+
+  const allValues = [...staticValues, ...keyframeValues].filter(
+    // value is string, and is unique
+    (value, index, self) =>
+      typeof value === 'string' && self.indexOf(value) === index,
+  ) as string[]
+
+  return allValues
+}
+
+export const createDefaultAssetStorageConfig = ({
+  project,
+  baseUrl = '',
+}: {
+  project: Project
+  baseUrl?: string
+}): IAssetStorageConfig => {
+  return {
+    coreAssetStorage: {
+      getAssetUrl: (assetId: string) => `${baseUrl}/${assetId}`,
+    },
+    createStudioAssetStorage: async () => {
+      // Check for support.
+      if (!('indexedDB' in window)) {
```

Comment:
might break ssr

Reply:
Hmm does SSR wait for async code? This does not run until studio is initialized. But yeah maybe better to be defensive about it. 😅

Reply:
haha yeah studio _could_ be initialized in ssr actually. only `studio.ui` is guaranteed not to be

Reply:
Alright, let's exercise an abundance of caution :D